tests/sandbox-lifecycle.test.ts

```diff
@@ -0,0 +1,135 @@
+import { describe, it, expect, beforeEach, afterEach } from 'vitest'
+import { CodeSandbox } from '../src/index'
+import { 
+  mockHibernateSuccess,
+  mockHibernateFailure,
+  mockStartVMSuccess,
+  mockShutdownSuccess,
+  mockShutdownFailure,
+  mockStartVMFailure,
+  setupTestEnvironment, 
+  cleanupTestEnvironment 
+} from './test-utils'
+
+describe('Sandbox lifecycle operations', () => {
+  beforeEach(() => {
+    setupTestEnvironment()
+  })
+
+  afterEach(() => {
+    cleanupTestEnvironment()
+  })
+
+  it('should successfully hibernate a sandbox', async () => {
+    const sandboxId = 'test-sandbox-hibernate'
+    
+    // Mock hibernate API call
+    const hibernateScope = mockHibernateSuccess(sandboxId)
+
+    const sdk = new CodeSandbox()
+    
+    // Test hibernate operation
+    await expect(sdk.sandboxes.hibernate(sandboxId)).resolves.not.toThrow()
```

Comment:
I think this is fine, but I'm curious if we should check for the actual response which I think contains `success: true`? Not sure if in the future it might not throw, but still error (and not respond with success true).

Reply:
So all these calls goes through `handleResponse`, which I honestly find very confusing. That we rely so much on the structure of the response, as opposed to the status code itself... but that's history for ya 😂 

But the only way `success` is in there, if there are related errors as well. So is no scenario of `success: true`, as the other payload for success has not "success" field... go figure